stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -200,18 +200,21 @@ public ProcessingStateMachine(
 
   private void skipRecord() {
     notifySkippedListener(currentRecord);
-    inProcessing = false;
+    markProcessingCompleted();
     actor.submit(this::readNextRecord);
     metrics.eventSkipped();
   }
 
   void readNextRecord() {
+    tryToReadNextRecord();
+  }
+
+  void markProcessingCompleted() {
```

Comment:
❓ Just to clarify:

The issue was that on each read next, which could happen in between the error loop, we might have reset the errorHandlingPhase. Wasn't this the case before then already? I'm wondering why this was an issue now.

Reply:
💭 Do we still need the count?! Can it be removed?

Reply:
>  Wasn't this the case before then already? 

Yes. This was already the case before. But it was not used for anything critical. It was more of an observability/debugging info.